controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -273,3 +284,85 @@ func (r *ClusterDeploymentReconciler) createImageRegistry(ctx context.Context, r
 	}
 	return nil
 }
+
+// setACICapabilities will set the install config override annotation in the AgentClusterInstall if there
+// are additional enabled capabilities that need to be defined.
+// For SNO (single node openshift) and non-baremetal platform MNO (multi-node openshift), this is set if
+// the OpenshiftAssistedControlPlane has additional capabilities set.
+// For MNO (multi-node openshift), this is set to the default list for baremetal platform,
+// but the OpenshiftAssistedControlPlane can specify additional capabilities to be appended to this list.
+func setACICapabilities(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) error {
+	isBaremetalPlatform := aci.Spec.PlatformType == hiveext.BareMetalPlatformType
+	if funk.IsEmpty(oacp.Spec.Config.Capabilities) && !isBaremetalPlatform {
+		return nil
+	}
+	baselineCapability, err := getBaselineCapability(oacp.Spec.Config.Capabilities.BaselineCapability, isBaremetalPlatform)
+	if err != nil {
+		return err
+	}
+
+	additionalEnabledCapabilities, err := getAdditionalCapabilities(oacp.Spec.Config.Capabilities.AdditionalEnabledCapabilities, isBaremetalPlatform)
+	if err != nil {
+		return err
+	}
+
+	installConfigOverride := fmt.Sprintf(`{"capabilities": {"baselineCapabilitySet": "%s", "additionalEnabledCapabilities": %s}}`, baselineCapability, additionalEnabledCapabilities)
+	if additionalEnabledCapabilities == "" {
+		// The user only specified BaselineCapability so no need to set additionalEnabledCapabilities in the override
+		installConfigOverride = fmt.Sprintf(`{"capabilities": {"baselineCapabilitySet": "%s"}}`, baselineCapability)
+	}
```

Comment:
nit: if we'd have an output struct to unmarshal capabilities in, the logic could be slightly simpler (omitempty for the additionalEnableCapabilities, no sprintfs, no conditional here)

Reply:
Good idea, tried to do it with a new type, wdyt?